build-aux/m4/ax_boost_thread.m4

```diff
@@ -71,6 +71,8 @@ AC_DEFUN([AX_BOOST_THREAD],
                  CXXFLAGS="-pthreads $CXXFLAGS"
              elif test "x$host_os" = "xmingw32" ; then
                  CXXFLAGS="-mthreads $CXXFLAGS"
+             elif test "x$host_os" = "xlinux-android" ; then
+                 CXXFLAGS="$CXXFLAGS"
```

Comment:
Does `-pthread` actually *fail* targeting Android?

Reply:
Yes it does, see [here](https://github.com/protocolbuffers/protobuf/issues/1373). I'm not sure why it worked before but right now building the master branch on Linux is broken.